pkg/metamanager/msg_processor.go

```diff
@@ -260,6 +260,8 @@ func (m *metaManager) processRemoteQuery(message model.Message) {
 		content, err := json.Marshal(resp.GetContent())
 		if err != nil {
 			log.LOGGER.Errorf("marshal remote query response content failed, %s", msgDebugInfo(&resp))
+			feedbackError(err, "Error to marshal message content", message, m.context)
```

Comment:
We should be a bit careful about `feedbackError` here. Any reason why should do this here? I think this PR is adding some UTs, does test codes rely on the error messages transported in the message channel?

Reply:
processRemoteQuery queries about a resource from cloud through websocket(EdgeHub). Hence if there is a marshall error in the content, we need to feedback to cloud. There is a Database error case below this. Since database save error is a local error, we are not sending feedback to cloud.

Reply:
Some of the UT cases rely on this change

Reply:
Is it possible to get rid of this rely?

Reply:
I feel it is OK to feedback the error in case of an unmarshal error. There are other instances in the code which do the same.